### PR TITLE
delete: only show when appropriate

### PR DESCRIPTION
**Pull request by patosullivan**
Fixes TLON-3076

Only allow posts to be deleted by their authors or the group admin.

### CODE REVIEW

File: packages/ui/src/components/ChatMessage/ChatMessageActions/MessageActions.tsx [hunk 1]
```diff
@@ -33,6 +33,16 @@ export default function MessageActions({
   const currentUserId = useCurrentUserId();
   const { addAttachment } = useAttachmentContext();
   const channel = useChannelContext();
+  const { data: group } = store.useGroup({ id: post.groupId ?? '' });
+  const currenUserIsAdmin = useMemo(
```
Comment (latter-bolden) [inline packages/ui/src/components/ChatMessage/ChatMessageActions/MessageActions.tsx:37]: We should really pull this out into something reusable. Looks like we do this same thing in 3 other places

**Review by latter-bolden — APPROVED** [review]
Makes sense! Up to you if ya wanna refactor the admin check into a helper func.

File: packages/ui/src/components/ChatMessage/ChatMessageActions/MessageActions.tsx (hunk 1 above)
Comment (patosullivan) [reply packages/ui/src/components/ChatMessage/ChatMessageActions/MessageActions.tsx:37]: Agreed, will do that now.

Comment (patosullivan) [reply packages/ui/src/components/ChatMessage/ChatMessageActions/MessageActions.tsx:37]: Wait, we *do* have a hook, we just don't use it consistently.

Comment (patosullivan) [reply packages/ui/src/components/ChatMessage/ChatMessageActions/MessageActions.tsx]: Ok, fixed here and in those other places.

Comment (latter-bolden) [reply packages/ui/src/components/ChatMessage/ChatMessageActions/MessageActions.tsx]: Awesome, ty!

**Pull request merged**